writer.go

```diff
@@ -49,28 +50,26 @@ func Open(paths ...string) (zapcore.WriteSyncer, func(), error) {
 func open(paths []string) ([]zapcore.WriteSyncer, func(), error) {
 	var openErr error
 	writers := make([]zapcore.WriteSyncer, 0, len(paths))
-	files := make([]*os.File, 0, len(paths))
+	closers := make([]io.Closer, 0, len(paths))
 	close := func() {
-		for _, f := range files {
-			f.Close()
+		for _, c := range closers {
+			c.Close()
 		}
 	}
 	for _, path := range paths {
-		switch path {
-		case "stdout":
-			writers = append(writers, os.Stdout)
-			// Don't close standard out.
+		if sink, err := NewSink(path); err == nil {
+			writers = append(writers, sink)
+			closers = append(closers, sink)
```

Comment:
I kept the two lists, `writers` and `closers`, because I didn't want to be too aggressive with the refactor and change the return value of 
`func open(paths []string) ([]zapcore.WriteSyncer, func(), error)`
to
`func open(paths []string) ([]Sink, func(), error)`

It would have rippled out to affect a few other funcs and we would have needed an array of `[]zapcore.WriteSyncer` for `CombineWriteSyncers` anyways.